README.md

```diff
@@ -37,143 +37,160 @@ pre-commit install
-# defined in the sequence
-job1 = {"runs": 20, "variables": {"omega_max": 6} }
-job2 = {"runs": 50, "variables": {"omega_max": 10.5} }
+#### Method 2: Username only
+If you only want to insert your username, but want a solution to have your password being secret,
+you can run the SDK without a password. A prompt will then ask for your password
 
-# Send the batch of jobs to the QPU and wait for the results
-batch = sdk.create_batch(serialized_sequence, [job1,job2], wait=True)
+```python
+sdk = SDK(username=username, project_id=project_id)
+```
 
-# You can also choose to run your batch on an emulator using the optional argument 'emulator'
-# For using a basic single-threaded QPU emulator that can go up to 10 qubits, you can specify the "EMU_FREE" emulator.
-from pasqal_cloud.device import EmulatorType
-batch = sdk.create_batch(serialized_sequence, [job1,job2], emulator=EmulatorType.EMU_FREE)
+#### Method 3: Use a custom token provider
+You can define a custom class to provide the token.
+For example, if you know your token, you can use that token to authenticate directly to our APIs as follows.
 
-# Once the QPU has returned the results, you can access them with the following:
-for job in batch.ordered_jobs:
-    print(job.result)
+```python
+class CustomTokenProvider(TokenProvider):
+    def get_token(self):
+        return "your-token" # Replace this value with your token
 
+
+sdk = SDK(token_provider=CustomTokenProvider(), project_id=project_id)
 ```
+Alternatively, create a custom TokenProvider that inherits from ExpiringTokenProvider. You should define a
+custom _query_token method which fetches your token. See Auth0TokenProvider implementation for an example.
 
-### Workload Creation
+### Create a batch of jobs
+
+The package main component is a python object called `SDK` which can be used to create a `Batch` and send it automatically
+to Pasqal APIs using an API token generated in the [user portal](https://portal.pasqal.cloud).
+
+A `Batch` is a group of jobs with the same sequence that will run on the same QPU. For each job of a given batch, you must set a value for each variable, if any, defined in your sequence.
+The batch sequence can be generated using [Pulser](https://github.com/pasqal-io/Pulser). See their [documentation](https://pulser.readthedocs.io/en/stable/),
+for more information on how to install the library and create your own sequence.
+
+The sequence should be a pulser Sequence object, once you have created it, you should serialize it as follows:
 
-You can create a workload through the SDK with the following command:
 ```python
-workload=sdk.create_workload(workload_type="<WORKLOAD_TYPE>",backend="<BACKEND>",config={"config_param_1":"value"})
+serialized_sequence = sequence.to_abstract_repr()
+```
 
-#You can cancel the workload by doing:
-sdk.cancel_workload(workload.id)
+Once you are login and have serialized your sequence, the last step is to define your jobs.
```

Comment:
Nit: This sentence doesn't make too much sense, we don't actually log in with the SDK and `once you are login` wouldn't be correct still

Reply:
I removed it